pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -93,11 +134,103 @@ func newStateImpl(checkpointDirectory, checkpointName string) state.State {
 
 // NewInMemoryManager returns an allocation manager that doesn't persist state.
 // For testing purposes only!
-func NewInMemoryManager() Manager {
-	return &manager{
-		allocated: state.NewStateMemory(nil),
-		actuated:  state.NewStateMemory(nil),
+func NewInMemoryManager(
+	statusManager status.Manager,
+	podResizeMutex *sync.Mutex,
+	canResizePod func(pod *v1.Pod) (bool, string, string),
+	triggerPodSync func(pod *v1.Pod)) Manager {
+	return newManager(
+		state.NewStateMemory(nil),
+		state.NewStateMemory(nil),
+		statusManager,
+		podResizeMutex,
+		canResizePod,
+		triggerPodSync,
+		true,
+	)
+}
+
+func newManager(allocated, actuated state.State,
+	statusManager status.Manager,
+	podResizeMutex *sync.Mutex,
+	canResizePod func(pod *v1.Pod) (bool, string, string),
+	triggerPodSync func(pod *v1.Pod),
+	skipPeriodicRetries bool) Manager {
+
+	m := &manager{
+		allocated:           allocated,
+		actuated:            actuated,
+		canResizePod:        canResizePod,
+		triggerPodSync:      triggerPodSync,
+		skipPeriodicRetries: skipPeriodicRetries,
+		statusManager:       statusManager,
+		podResizeMutex:      podResizeMutex,
+		pendingResizeChan:   make(chan struct{}, 1),
+	}
+
+	return m
+}
+
+func (m *manager) Run() {
+	// Returns a list of successful retried pods.
+	retryPendingResizes := func() []*v1.Pod {
```

Comment:
nit: make this a full method, rather than an anonymous function (shows up better in stack traces, and easier to read)